PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/transaction_layer.cpp

```diff
@@ -0,0 +1,32 @@
+#include "transaction_layer.h"
+#include "Tlp.h"
+/**
+ * @brief Checks the gate equation to determine if a TLP can be transmitted.
+ * @param sharedCreditLimit An array of shared credit limits.
+ *							Example: sharedCreditLimit[0] = CplDSharedCreditLimit
+ *									 sharedCreditLimit[1] = CplHSharedCreditLimit
+ * @param shardCreditConsumed An array of shared credit limits.
+ *							  Example: shardCreditConsumed[0] = CplDSharedCreditConsumed
+ *								       shardCreditConsumed[1] = CplHSharedCreditConsumed
+ * @param tlp The TLP (Transaction Layer Packet) object being transmitted.
+ * @return True if the gate equation is valid, false otherwise.
+*/
+bool TransactionLayer::checkGateEquation(int sharedCreditLimit[2], int shardCreditConsumed[2], Tlp tlp) {
+	// checking for the Data gate equation
+	int dataSharedCreditLimit = sharedCreditLimit[0];
+	int dataSharedCreditConsumed = shardCreditConsumed[0];
+	int twoPowCreditDataFieldSize = pow(CREDIT_DATA_FIELD_SIZE, 2);
+	
+	if ((dataSharedCreditLimit - (dataSharedCreditConsumed + tlp.dataConsumption) % twoPowCreditDataFieldSize) % twoPowCreditDataFieldSize > twoPowCreditDataFieldSize / 2)
```

Comment:
We will wait till the TLP class is done before merging this to make sure whether dataConsumption is an attribute or a method

Reply:
@MMostafa-Hub Can you please rebase and check the project is building so we can merge?